source/Applications/Advanced/TransformViaArucoMarker/TransformViaArucoMarker.cpp

```diff
@@ -0,0 +1,315 @@
+/*
+This example shows how to transform a point cloud using the pose of an aruco marker.
+*/
+
+#include <Zivid/Experimental/Calibration.h>
+#include <Zivid/Visualization/Visualizer.h>
+#include <Zivid/Zivid.h>
+
+#include <algorithm>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <aruco_nano.h>
+#include <cmath>
+#include <iostream>
+
+namespace
+{
+    struct Line2d
+    {
+        double slope;
+        double intercept;
+    };
+
+    int clamp(const int n, const int lower, const int upper)
+    {
+        return std::max(lower, std::min(n, upper));
+    }
+
+    // Fit a line y=a*x + b to 2 points
+    Line2d fitLine(const cv::Point2d &point1, const cv::Point2d &point2)
+    {
+        const auto slope{ (point2.y - point1.y) / (point2.x - point1.x) };
+        const auto intercept{ point1.y - slope * point1.x };
+        return { slope, intercept };
+    };
+
+    cv::Mat pointCloudToCvBGR(const Zivid::PointCloud &pointCloud)
+    {
+        auto rgb = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); // NOLINT(hicpp-signed-bitwise)
+        pointCloud.copyData(
+            reinterpret_cast<Zivid::ColorRGBA *>(rgb.data)); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
+        auto bgr = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); // NOLINT(hicpp-signed-bitwise)
+        cv::cvtColor(rgb, bgr, cv::COLOR_RGBA2BGR);
+
+        return bgr;
+    }
+
+    std::vector<cv::Point2f> estimate2DMarkerCenters(const aruconano::Marker &markers)
+    {
+        if(markers.empty())
+        {
+            std::cout << "No markers detected";
+            return {};
+        }
+
+        std::vector<cv::Point2f> markerCenters2D;
+        markerCenters2D.reserve(markers.size());
+
+        const auto corner0 = markers[0];
+        const auto corner1 = markers[1];
+        const auto corner2 = markers[2];
+        const auto corner3 = markers[3];
+
+        // Fit line between two diagonal corners
+        const auto backDiagonal{ fitLine(corner2, corner0) };
+        const auto forwardDiagonal{ fitLine(corner3, corner1) };
```

Comment:
Why 'forward' and 'back' diagonal?

Reply:
One diagonal goes from left to right (if you picture the 4 corners in front of you), while the other diagonal goes from right to left. I am not able to come up with better names for them:/

Reply:
diagonal02 and diagonal 13? since you use corners 0 1 2 3? Or diagonalCorner0ToCorner2 - these are some ideas to start with. I think 'back' and 'forward' doesn't reveal anything.